controllers/k8ssandra/datacenters.go

```diff
@@ -126,23 +126,39 @@ func (r *K8ssandraClusterReconciler) reconcileDatacenters(ctx context.Context, k
 				}
 			}
 
-			if !cassandra.DatacenterReady(actualDc) {
-				logger.Info("Waiting for datacenter to become ready")
+			if actualDc.Status.CassandraOperatorProgress != cassdcapi.ProgressReady {
+				logger.Info("Waiting for datacenter to finish updating")
 				return result.RequeueSoon(r.DefaultDelay), actualDcs
 			}
 
-			logger.Info("The datacenter is ready")
+			if actualDc.Spec.Stopped {
+				if !cassandra.DatacenterStopped(actualDc) {
+					logger.Info("Waiting for datacenter to satisfy Stopped condition")
+					return result.RequeueSoon(r.DefaultDelay), actualDcs
+				}
+			} else {
+				if !cassandra.DatacenterReady(actualDc) {
+					logger.Info("Waiting for datacenter to satisfy Ready condition")
+					return result.RequeueSoon(r.DefaultDelay), actualDcs
+				}
+			}
+
+			logger.Info("The datacenter is reconciled")
 
 			actualDcs = append(actualDcs, actualDc)
 
-			if recResult := r.checkSchemas(ctx, kc, actualDc, remoteClient, logger); recResult.Completed() {
-				return recResult, actualDcs
-			}
+			if !actualDc.Spec.Stopped {
 
-			if annotations.HasAnnotationWithValue(kc, api.RebuildDcAnnotation, dcKey.Name) {
-				if recResult := r.reconcileDcRebuild(ctx, kc, actualDc, remoteClient, logger); recResult.Completed() {
+				if recResult := r.checkSchemas(ctx, kc, actualDc, remoteClient, logger); recResult.Completed() {
 					return recResult, actualDcs
 				}
+
+				if annotations.HasAnnotationWithValue(kc, api.RebuildDcAnnotation, dcKey.Name) {
+					if recResult := r.reconcileDcRebuild(ctx, kc, actualDc, remoteClient, logger); recResult.Completed() {
```

Comment:
@jsanda if a dc is added in stopped state, should we call `reconcileDcRebuild` for it?

Reply:
We should throw a validation error in that case. cass-operator will not finish building the new DC if it's in stopped state.

Reply:
Hm that's crazy indeed. I just created a new cassdc in stopped state, and cass-operator happily created the service and even the empty statefulsets for me. No validation errors. Conditions Initialized and Stopped are true. This is not going to work, I will add a validation error to forbid creating a dc in stopped state.